src/mongocxx/test/uri.cpp

```diff
@@ -69,5 +73,273 @@ TEST_CASE("URI", "[uri]") {
             REQUIRE(e.code() == mongocxx::error_code::k_invalid_uri);
         }
     }
+
+    SECTION("Multiple hosts") {
+        mongocxx::uri uri{"mongodb://host1:123,host2:456"};
+        REQUIRE(uri.hosts().size() == 2);
+        REQUIRE(uri.hosts()[0].name == "host1");
+        REQUIRE(uri.hosts()[0].port == 123);
+        REQUIRE(uri.hosts()[1].name == "host2");
+        REQUIRE(uri.hosts()[1].port == 456);
+    }
+
+    SECTION("Getter is case insensitive") {
+        mongocxx::uri uri = mongocxx::uri{"mongodb://host/?cOnNeCtTiMeOuTmS=123"};
+        REQUIRE(uri.connect_timeout_ms());
+        std::int32_t connect_timeout_ms = *uri.connect_timeout_ms();
+        REQUIRE(123 == connect_timeout_ms);
+    }
 }
+
+template <typename fn>
+static void _test_string_option(std::string optname, fn getter_fn) {
+    /* Not present. */
+    mongocxx::uri uri{"mongodb://host/"};
+    REQUIRE(!getter_fn(uri));
+
+    /* Present. */
+    uri = mongocxx::uri{"mongodb://host/?" + optname + "=abc"};
+    REQUIRE(getter_fn(uri));
+    REQUIRE(0 == getter_fn(uri)->compare("abc"));
+
+    /* Present but empty. */
+    uri = mongocxx::uri{"mongodb://host/?" + optname + "=&connectTimeoutMS=123"};
+    REQUIRE(getter_fn(uri));
+    REQUIRE(0 == getter_fn(uri)->compare(""));
+}
+
+template <typename fn>
+static void _test_bool_option(std::string optname, fn getter_fn) {
+    /* Not present. */
+    mongocxx::uri uri{"mongodb://host/"};
+    REQUIRE(!getter_fn(uri));
+
+    /* Present. */
+    uri = mongocxx::uri{"mongodb://host/?" + optname + "=true"};
+    REQUIRE(getter_fn(uri));
+    REQUIRE(true == *getter_fn(uri));
+
+    /* Present, but false. */
+    uri = mongocxx::uri{"mongodb://host/?" + optname + "=false"};
+    REQUIRE(getter_fn(uri));
+    REQUIRE(false == *getter_fn(uri));
+
+    /* Present but empty. libmongoc considers it unset. */
+    uri = mongocxx::uri{"mongodb://host/?" + optname + "="};
+    REQUIRE(!getter_fn(uri));
+
+    /* Present, but numeric. */
+    uri = mongocxx::uri{"mongodb://host/?" + optname + "=1"};
```

Comment:
All numeric values would result in `true` right?

Reply:
I actually wasn't sure off hand. libmongoc actually only interprets a few numbers as boolean values:
https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/src/mongoc/mongoc-uri.c#L1097-L1111

Apparently `-1` is considered false...

I added a test for good measure.